lightyear/src/transport/mod.rs

```diff
@@ -72,8 +77,50 @@ pub trait PacketReceiver: Send + Sync {
     fn recv(&mut self) -> Result<Option<(&mut [u8], SocketAddr)>>;
 }
 
+impl PacketSender for Box<dyn PacketSender> {
+    fn send(&mut self, payload: &[u8], address: &SocketAddr) -> Result<()> {
+        (**self).send(payload, address)
+    }
+}
+
+// impl<T: PacketSender> PacketSender for &mut T {
+//     fn send(&mut self, payload: &[u8], address: &SocketAddr) -> Result<()> {
+//         (*self).send(payload, address)
+//     }
+// }
+
 impl PacketReceiver for Box<dyn PacketReceiver> {
     fn recv(&mut self) -> Result<Option<(&mut [u8], SocketAddr)>> {
         (**self).recv()
     }
 }
+
+// impl<T: PacketReceiver> PacketReceiver for &mut T {
+//     fn recv(&mut self) -> Result<Option<(&mut [u8], SocketAddr)>> {
+//         (*self).recv()
+//     }
+// }
+
+impl PacketSender for &mut dyn PacketSender {
+    fn send(&mut self, payload: &[u8], address: &SocketAddr) -> Result<()> {
+        (*self).send(payload, address)
+    }
+}
+
+impl PacketSender for Box<&mut dyn PacketSender> {
+    fn send(&mut self, payload: &[u8], address: &SocketAddr) -> Result<()> {
+        (**self).send(payload, address)
+    }
+}
+
+impl PacketReceiver for &mut dyn PacketReceiver {
+    fn recv(&mut self) -> Result<Option<(&mut [u8], SocketAddr)>> {
+        (*self).recv()
+    }
+}
+
+impl PacketReceiver for Box<&mut dyn PacketReceiver> {
+    fn recv(&mut self) -> Result<Option<(&mut [u8], SocketAddr)>> {
```

Comment:
`Result<Option<...>>` looks a little weird to me

Reply:
It's because I want to differentiate between:
- error: io error
- Ok(None): no error, but no packets
- Ok(Some()): no error, and there is a packet.

(this part did not change btw)